### PR TITLE
Document address function calls

### PR DESCRIPTION
**Pull request by naftali-g**
Naming convention:
 - PRs for features that are in design should have the "proposal" label
 - PRs for features that haven't landed yet should have the "future" label
 - PRs for upcoming releases should have the "release" label
 - PRs with new documentation for existing features should have the "existing feature" label

Before requesting review:
 - Make sure there is a ticket in the DOC board in Jira
 - Make sure CI is passing
   - Spell check failure may require adding backticks around code or updating `spelling_wordlist.txt`
   - See `README.md` for information about style and markdown syntax
   - If the CI Details link gives a 404, you need to log in to readthedocs.com
 - Add link to generated documentation here
   - you can find this by following the read the docs link from the CI check
 - Ask for help in #documentation

Jira ticket: TODO
Link to generated documentation: TODO

### CODE REVIEW

File: docs/cvl/expr.md [hunk 1]
```diff
@@ -365,6 +365,15 @@ while verifying the rule, and will provide a separate verification report for
...
 {term}`parametric rule`s.
 
+Another possible way to have the Prover consider options for a function is by
+prefixing the function name with an `address` typed variable. In this case the
+Prover will consider every possible contract in the scene that implements a
+function that matches the signature provided by the call (if no such function
+exists in the scene the prover will fail with an error). If there are no
+relevant constraints on the `address` variable the prover may also consider the
+case that its value does not match any contract in the scene and in this case
+the Prover will consider this a violation of the rule.
```
Comment (johspaeth) [inline docs/cvl/expr.md:375]: Can we add an example?

```suggestion
Another possible way to have the Prover consider options for a function is by
prefixing the function name with an `address` typed variable. For instance,
you can write a rule such as `rule r (env e; address a){assert a.balanceOf(e) > 0;}`.
In this case the Prover will consider every possible contract in the scene that 
implements a function that matches the signature provided by the call 
(if no such function exists in the scene the prover will fail with an error). If there 
are no relevant constraints on the `address` variable the prover may also consider 
the case that its value does not match any contract in the scene and in this case
the Prover will consider this a violation of the rule.
```

**Review by johspaeth — CHANGES_REQUESTED** [review]
Can we please add a minimal example here?

**Review by johspaeth — APPROVED** [review]
Ok to merge. We'll _not_ include an example for now.

File: docs/cvl/expr.md [hunk 2]
```diff
@@ -365,6 +365,15 @@ while verifying the rule, and will provide a separate verification report for
 each checked method.  Rules that use this feature are referred to as
 {term}`parametric rule`s.
 
+Another possible way to have the Prover consider options for a function is by
+prefixing the function name with an `address` typed variable. In this case the
+Prover will consider every possible contract in the scene that implements a
+function that matches the signature provided by the call (if no such function
+exists in the scene the prover will fail with an error). If there are no
+relevant constraints on the `address` variable the prover may also consider the
```
Comment (jar-ben) [inline docs/cvl/expr.md:373]: _"the prover may also consider"_? Either it considers it or not, do not use "may"

Comment (jar-ben) [inline docs/cvl/expr.md:373]: _"If there are no relevant constraints"_. What do you mean by relevant constraints?

**Pull request merged**